boofuzz/sessions.py

```diff
@@ -365,13 +370,20 @@ def __init__(self, session_filename=None, index_start=1, index_end=None, sleep_t
         self.restart_interval = restart_interval
         self.web_port = web_port
         self._keep_web_open = keep_web_open
+        self.console_gui = console_gui
         self._crash_threshold_node = crash_threshold_request
         self._crash_threshold_element = crash_threshold_element
         self.restart_sleep_time = restart_sleep_time
         if fuzz_data_logger is not None:
             raise exception.BoofuzzError('Session fuzz_data_logger is deprecated. Use fuzz_loggers instead!')
         if fuzz_loggers is None:
-            fuzz_loggers = [fuzz_logger_text.FuzzLoggerText()]
+            if self.console_gui and os.name != 'nt':
+                default_signal_handler = signal.getsignal(signal.SIGWINCH)
+                fuzz_loggers = [fuzz_logger_curses.FuzzLoggerCurses(web_port=self.web_port)]
+                self._keep_web_open = False
+                signal.signal(signal.SIGWINCH, default_signal_handler)
```

Comment:
I realize why it was bugging out for me -- I was manually creating a `FuzzLoggerCurses()` and passing it into the Session class without configuring all of these.

What if we move the two signal handler lines into the FuzzLoggerCurses constructor? A quick test worked for me with these lines surrounding the thread start.

Reply:
> What if we move the two signal handler lines into the FuzzLoggerCurses constructor? A quick test worked for me with these lines surrounding the thread start.

That totally makes sense, I will do some testing and move it over.

> 
> 
> A couple review comments below. This latest commit didn't run for me without a modification -- I suspect it has to do with some of the Py2/Py3 changes. Here's the patch:
> 
> ```
> > git diff
> diff --git a/boofuzz/pedrpc.py b/boofuzz/pedrpc.py
> index d4d9fa1..92c9ca3 100644
> --- a/boofuzz/pedrpc.py
> +++ b/boofuzz/pedrpc.py
> @@ -9,7 +9,7 @@ import select
>  from . import exception
>  
>  
> -class Client:
> +class Client(object):
> ```

Is this patch still required when using the session option to construct the curses-class?

Reply:
I didn't try specifically with the Session option but without this patch -- I can make it a separate PR too.

Reply:
@SR4ven by the way can you explain a bit about what those signal handling lines do? I checked the the [docs for signal](https://docs.python.org/2/library/signal.html#signal.getsignal) but I don't understand why they make things work here.

Reply:
@jtpereyda So, `default_signal_handler = signal.getsignal(signal.SIGWINCH)` saves the current SIGWINCH handler, `signal.signal(signal.SIGWINCH, default_signal_handler)` restores it. [Reference](https://stackoverflow.com/questions/22916783/reset-python-sigint-to-default-signal-handler)
The position of the default_signal_handler saving  after ac4afff might be a little too far down in the code, moving it above `self._stdscr = curses.initscr()` would be safer, but it passed my tests as it is. 

This is necessary because curses sets up a custom handler. That handler allows interrupting the program at any point, messing with the timings and causing errors.
The default signal handler does literally nothing, so we just use that and check for a terminal size change periodically. Not the most responsive way, but it works.

I hope that makes it more clear.